src/_pytest/debugging.py

```diff
@@ -88,25 +89,33 @@ def set_trace(cls, set_break=True):
                 capman.suspend_global_capture(in_=True)
             tw = _pytest.config.create_terminal_writer(cls._config)
             tw.line()
-            if capman and capman.is_globally_capturing():
-                tw.sep(">", "PDB set_trace (IO-capturing turned off)")
-            else:
-                tw.sep(">", "PDB set_trace")
+            if cls._recursive == 0:
```

Comment:
Maybe use `<= 0` here, and have a warning if it is `< 0`?  If so, how?  Using `pytest.warn`?

Reply:
(`pytest.warns` checks for warnings in a `with` block, I assume you meant `warnings.warn`)

Not sure if it would be helpful here, because warnings would only be printed in the warnings summary at the end of the session. Perhaps just writing a warning message to the `tw` directly `Internal '_recursive' flag is {}, please report to the pytest team at https...`?

Reply:
Left this out for now.